Marlin/UBL_G29.cpp

```diff
@@ -712,58 +712,193 @@
+          lcd_quick_feedback();
+          STOW_PROBE();
+          while (ubl_lcd_clicked() ) {
+            idle();
+          }
+          ubl_has_control_of_lcd_panel = false;
+          restore_ubl_active_state_and_leave();
+          safe_delay(50);  // Debounce the Encoder wheel
+          return;
         }
-        ubl_has_control_of_lcd_panel = false;
-        restore_ubl_active_state_and_leave();
-        safe_delay(50);  // Debounce the Encoder wheel
-        return;
-      }
 
-      location = find_closest_mesh_point_of_type(INVALID, lx, ly, 1, NULL, do_furthest );  // the '1' says we want the location to be relative to the probe
-      if (location.x_index >= 0 && location.y_index >= 0) {
-        const float xProbe = ubl.map_x_index_to_bed_location(location.x_index),
-                    yProbe = ubl.map_y_index_to_bed_location(location.y_index);
-        if (xProbe < MIN_PROBE_X || xProbe > MAX_PROBE_X || yProbe < MIN_PROBE_Y || yProbe > MAX_PROBE_Y) {
-          SERIAL_PROTOCOLLNPGM("?Error: Attempt to probe off the bed.");
-          ubl_has_control_of_lcd_panel = false;
-          goto LEAVE;
+        location = find_closest_mesh_point_of_type(INVALID, lx, ly, 1, NULL, do_furthest );  // the '1' says we want the location to be relative to the probe
+        if (location.x_index >= 0 && location.y_index >= 0) {
+          const float xProbe = ubl.map_x_index_to_bed_location(location.x_index),
+                      yProbe = ubl.map_y_index_to_bed_location(location.y_index);
+          if (xProbe < MIN_PROBE_X || xProbe > MAX_PROBE_X || yProbe < MIN_PROBE_Y || yProbe > MAX_PROBE_Y) {
+            SERIAL_PROTOCOLLNPGM("?Error: Attempt to probe off the bed.");
+            ubl_has_control_of_lcd_panel = false;
+            goto LEAVE;
+          }
+          const float measured_z = probe_pt(xProbe, yProbe, stow_probe, g29_verbose_level);
+          z_values[location.x_index][location.y_index] = measured_z + zprobe_zoffset;
         }
-        const float measured_z = probe_pt(xProbe, yProbe, stow_probe, g29_verbose_level);
-        z_values[location.x_index][location.y_index] = measured_z + zprobe_zoffset;
-      }
 
-      if (do_ubl_mesh_map) ubl.display_map(map_type);
+        if (do_ubl_mesh_map) ubl.display_map(map_type);
 
-    } while (location.x_index >= 0 && location.y_index >= 0);
+      } while (location.x_index >= 0 && location.y_index >= 0);
 
-    LEAVE:
+      LEAVE:
 
-    STOW_PROBE();
-    restore_ubl_active_state_and_leave();
+      STOW_PROBE();
+      restore_ubl_active_state_and_leave();
+
+      do_blocking_move_to_xy(
+        constrain(lx - (X_PROBE_OFFSET_FROM_EXTRUDER), X_MIN_POS, X_MAX_POS),
+        constrain(ly - (Y_PROBE_OFFSET_FROM_EXTRUDER), Y_MIN_POS, Y_MAX_POS)
+      );
+    }
+
+  #elif ENABLED(UBL_RECTANGULAR_PROBING)
+
+    void probe_entire_mesh(const float &lx, const float &ly, const bool do_ubl_mesh_map, const bool stow_probe, bool do_furthest) {

```

Comment:
Unless the function signature is different between the different styles of probing, the code should be structured like:
```cpp
void probe_entire_mesh(const float &lx, const float &ly, const bool do_ubl_mesh_map, const bool stow_probe, bool do_furthest) {
  // common init code here

  #if ENABLED(UBL_CIRCULAR_PROBING)
    // . . . circular probing
  #elif ENABLED(UBL_RECTANGULAR_PROBING)
    // . . . rectangle probing
  #endif

  // common closing code here
}
```